### PR TITLE
✨(frontend) refetch user profile after edition

### PR DESCRIPTION
**Pull request by rlecellier**
On click on modify profile, we open a new window in OpenEdx profile page.
When modification are done, we would like to refresh informations display on this page.

This onWindowFocus event will be trigger the first time the user came back from OpenEdx profile page to refresh the informations. /!\ It's not perfect as it'll not work it he go and back from one tab to the other.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -20,6 +20,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## Fixed
 
+- User profile in the learner dashboard is now always synchornized with
```
Comment (jbpenrath) [inline CHANGELOG.md:23]:
```suggestion
- User profile in the learner dashboard is now always synchronized with 
```

File: src/frontend/js/pages/DashboardOpenEdxProfile/index.tsx [hunk 2]
```diff
@@ -111,7 +111,24 @@ export const DEFAULT_DISPLAYED_FORM_VALUE = ' - ';
...
+  };
+
+  // Because the user can leaver the page to update his profile in openEdx.
+  // if we invalidate before leaving the page, it's directly refetched.
+  // We need to invalidate openEdx's profile each time we reload this page.
+  // it's the only way to have user data synchronised between richie's frontend
+  // and openEdx's database.
+  useEffect(() => {
+    invalidate();
+  }, []);
```
Comment (jbpenrath) [inline src/frontend/js/pages/DashboardOpenEdxProfile/index.tsx:131]: Did you try to trigger invalidation on `beforeunload` event emits by window ?
```suggestion
  const onClickModifyProfile = async () => {
    window.addEventListener('beforeunload', invalidate, { once : true });
    window.location.replace(`${context.authentication.endpoint}/account/settings`);
  };
```

Comment (rlecellier) [reply src/frontend/js/pages/DashboardOpenEdxProfile/index.tsx:131]: I tried multiple new solution including this beforeunload.
This event do the same as other ways to trigger invalidate, it invalidate then refetch.
I tried to use `removeQueries` and/or `resetQueries` that execute when the query "enabled" switch to `false`.
It work without the `location.replace`: i trigger the reset, no refresh happened, i change route then comme back, the data is refetch.
But it didn't work with the `location.replace`, the reset is trigger, i change page, edit the profile, come back to richie home then profile -> it found the data from somewhere and don't trigger a new fetch.

File: src/frontend/js/index.tsx [hunk 3]
```diff
@@ -102,7 +102,7 @@ async function render() {
     document.body.append(rootContainer);
 
     // React-query
-    const queryClient = createQueryClient({ logger: true, persister: true });
+    const queryClient = createQueryClient({ logger: true, persister: false });
```
Comment (jbpenrath) [inline src/frontend/js/index.tsx:105]: Debug residue ?

File: src/frontend/js/pages/DashboardCreditCardsManagement/DashboardEditCreditCard.spec.tsx [hunk 4]
```diff
@@ -97,6 +97,7 @@ describe('<DahsboardEditCreditCard/>', () => {
   });
 
   it('updates a credit card name', async () => {
+    fetchMock.get('https://endpoint.test/api/v1.0/user/me', richieUser);
```
Comment (jbpenrath) [inline src/frontend/js/pages/DashboardCreditCardsManagement/DashboardEditCreditCard.spec.tsx:100]: You could add it in the `beforeEach` nope ?

File: src/frontend/js/pages/DashboardOpenEdxProfile/index.spec.tsx [hunk 5]
```diff
@@ -32,9 +32,11 @@ describe('pages.DashboardOpenEdxProfile', () => {
   setupJoanieSession();
 
   it('should render profile informations', async () => {
+    const richieUser = UserFactory().one();
+    fetchMock.get('https://endpoint.test/api/v1.0/user/me', richieUser);
```
Comment (jbpenrath) [inline src/frontend/js/pages/DashboardOpenEdxProfile/index.spec.tsx:36]: Ditto ?

File: src/frontend/js/pages/DashboardOpenEdxProfile/index.tsx [hunk 6]
```diff
@@ -225,13 +231,9 @@ const DashboardOpenEdxProfile = () => {
...
       <Form.Footer>
-        <RouterButton
-          fullWidth
-          target="_blank"
-          href={`${context.authentication.endpoint}/account/settings`}
-        >
+        <Button fullWidth onClick={onClickModifyProfile}>
           <FormattedMessage {...messages.editButtonLabel} />
-        </RouterButton>
+        </Button>
```
Comment (jbpenrath) [inline src/frontend/js/pages/DashboardOpenEdxProfile/index.tsx:236]: For a11y purpose, I think we could use href attribute to generate a link.
Does following suggestion work ?

```suggestion
        <Button
          fullWidth
          href={`${context.authentication.endpoint}/account/settings`}
          onClick={clearReactQueryCache}
        >
          <FormattedMessage {...messages.editButtonLabel} />
        </Button>
```

File: src/frontend/js/pages/DashboardCreditCardsManagement/DashboardEditCreditCard.spec.tsx (hunk 4 above)
Comment (rlecellier) [reply src/frontend/js/pages/DashboardCreditCardsManagement/DashboardEditCreditCard.spec.tsx:100]: it's not needed everywhere (not on the last test).

**Review by jbpenrath — APPROVED** [review]

**Pull request merged**